server/src/main/java/com/hedera/block/server/consumer/ConsumerStreamResponseObserver.java

```diff
@@ -118,7 +118,17 @@ public void onEvent(
 
                 final SubscribeStreamResponseUnparsed subscribeStreamResponse = event.get();
                 final ResponseSender responseSender = getResponseSender(subscribeStreamResponse);
-                responseSender.send(subscribeStreamResponse);
+                try {
+                    responseSender.send(subscribeStreamResponse);
+                } catch (IllegalArgumentException e) {
+                    throw e;
+                } catch (RuntimeException e) {
+                    isResponsePermitted.set(false);
+                    subscriptionHandler.unsubscribe(this);
+                    LOGGER.log(
+                            DEBUG,
+                            "RuntimeException caught from Pipeline instance. Unsubscribed ConsumerBlockItemObserver instance");
```

Comment:
should this be DEBUG? I get is not a actual `error` but maybe a warning?

Reply:
The exceptions we're catching here will be a downstream client hitting control-c 99% of the time

Reply:
I added more fine-grained exception handling. Now the UncheckedIOExceptions will get logged to DEBUG and any other RuntimeException should show up as an ERROR.